pkg/codegen/docs.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// docFileName defines the name of the files that will be aggregated into overall docs
+const docFileExtension = ".md"
+
+type docFile struct {
+	content  []byte
+	resource string
+	group    string
+	version  string
+}
+
+func generateDocs(resourcesBaseDir, outputFilePath string) error {
+	outputFile, err := os.OpenFile(outputFilePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		return err
+	}
```

Comment:
I did something like this - I also merged/wrapped the original error so that you get both of them if you have two errors. It's not super pretty but I think it will get the job done.